content/tutorial-air-quality-analysis.md

```diff
@@ -0,0 +1,360 @@
+worst affected cities by air pollution, before and during the lockdown from March to June 2020. For this, we will first compute
+the Air Quality Index for each hour from the collected pollutant measurements. Next, we will sample these indices and perform
+a [paired Student's t-test](https://en.wikipedia.org/wiki/Student%27s_t-test#Dependent_t-test_for_paired_samples) on them. It will statistically show us that the air quality improved due to the lockdown, supporting our intuition.
+
+Let's start by importing the necessary libraries into our environment. Also, enable IPython's [magic function](https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-matplotlib) to see the Matplotlib plots in the cell outputs.
+
+```{code-cell} ipython3
+import numpy as np
+from scipy import stats
+import matplotlib.pyplot as plt
+
+%matplotlib inline
+```
+
+## Building the dataset
+
+We will use a condensed version of the [Air Quality Data in India](https://www.kaggle.com/rohanrao/air-quality-data-in-india) dataset. This dataset contains air quality data and AQI (Air Quality Index) at hourly and daily level of various stations across multiple cities in India. The condensed version available with this tutorial contains hourly pollutant measurements for Delhi
+from May 31, 2019 to June 30, 2020. It has measurements of the standard pollutants that are required for Air Quality Index calculation and a few other important ones:
+Particulate Matter (PM 2.5 and PM 10), nitrogen dioxide (NO2), ammonia (NH3), sulfur dioxide (SO2), carbon monoxide (CO), ozone (O3), oxides of nitrogen (NOx), nitric oxide (NO), benzene, toluene, and xylene.
+
+Let's print out the first few rows to have a glimpse of our dataset.
+
+```{code-cell} ipython3
+! head air-quality-data.csv
+```
+
+For the purpose of this tutorial, we are only concerned with standard pollutants required for calculating the AQI, viz., PM 2.5, PM 10, NO2, NH3, SO2, CO, and O3. So, we will only import these particular columns with [np.genfromtxt](https://numpy.org/devdocs/reference/generated/numpy.genfromtxt.html). We'll initially create two sets: `pollutants_A` with PM 2.5, PM 10, NO2, NH3, and SO2, and `pollutants_B` with CO and O3. The
+two sets will be processed slightly differently, as we'll see later on.
+
+```{code-cell} ipython3
+pollutants_A = np.genfromtxt("air-quality-data.csv", dtype='f8', delimiter=",",
+                             skip_header=1, usecols=(1, 2, 3, 4, 5))
+pollutants_B = np.genfromtxt("air-quality-data.csv", dtype='f8', delimiter=",",
+                             skip_header=1, usecols=(6, 7))
+
+print(pollutants_A.shape)
+print(pollutants_B.shape)
+```
+
+Our dataset might contain missing values, denoted by `NaN`, so let's do a quick check with [np.isnan](https://numpy.org/devdocs/reference/generated/numpy.isnan.html).
+
+```{code-cell} ipython3
+print(np.where(np.isnan(pollutants_A) == True))
+print(np.where(np.isnan(pollutants_B) == True))
```

Comment:
Just a suggestion, feel free to ignore:

```suggestion
np.all(np.isfinite(pollutants_A)) and np.all(np.isfinite(pollutants_B))
```
Or, better yet, if you do decide to go with the `pollutant_data` approach, you can just run `np.all(np.isfinite(pollutant_data))`